src/components/Compare/CompareTable.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { useState } from 'react';
+import { TableBody } from '@material-ui/core';
+import {
+  Wrapper,
+  StyledTable,
+  ArrowContainer,
+  TableHeadContainer,
+  Row,
+  Cell,
+  Footer,
+  ViewAllLink,
+  Header,
+} from 'components/Compare/Compare.style';
+import CompareTableRow from 'components/Compare/CompareTableRow';
+import HeaderCell from 'components/Compare/HeaderCell';
+import { getLocationNames, Location } from 'common/locations';
+import {
+  stateSummary,
+  countySummary,
+  LocationSummary,
+} from 'common/location_summaries';
+import { Metric } from 'common/metric';
+import ExpandLessIcon from '@material-ui/icons/ExpandLess';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import { COLOR_MAP } from 'common/colors';
+import { fail } from 'common/utils';
+
+const locations: any = getLocationNames();
+
+export interface SummaryForCompare {
+  locationInfo: Location;
+  metricsInfo: LocationSummary;
+}
+
+const CompareTable = (props: {
+  stateId?: string;
+  stateName?: string;
+  county?: any | null;
+  setShowModal: any;
+  isModal?: Boolean;
+  locationsViewable?: number;
+  isLocationPage?: Boolean;
+  isHomepage?: Boolean;
+}) => {
+  const [sorter, setSorter] = useState(5);
+  const [sortDescending, setSortDescending] = useState(true);
+
+  const orderedMetrics = [
+    Metric.CASE_DENSITY,
+    Metric.CASE_GROWTH_RATE,
+    Metric.POSITIVE_TESTS,
+    Metric.HOSPITAL_USAGE,
+    Metric.CONTACT_TRACING,
+  ];
+
+  const statesArr = locations
+    .filter((location: Location) => !location.county)
+    .map((stateInfo: Location) => {
+      if (stateInfo.state_code) {
+        return {
+          locationInfo: stateInfo,
+          metricsInfo: stateSummary(stateInfo.state_code),
+        };
+      } else {
+        fail('No state code');
+      }
+    });
```

Comment:
I think it would be better to pass this information as param for the component

Reply:
moved these functions to common/utils/compare